WalletWasabi.Gui/ViewModels/StatusBarViewModel.cs

```diff
@@ -253,6 +257,28 @@ public void Initialize(NodesCollection nodes, WasabiSynchronizer synchronizer)
 
 					UpdateAvailable = !x.ClientUpToDate;
 					CriticalUpdateAvailable = !x.BackendCompatible;
+
+					try
+					{
+						if (Global.LegalDocuments is null || Global.LegalDocuments.Version != x.LegalDocumentsVersion)
+						{
+							try
+							{
+								var legalResp = await LegalDocuments.FetchLatestAsync(Global.DataDir, () => Global.Config.GetCurrentBackendUri(), Global.Config.TorSocks5EndPoint, CancellationToken.None);
```

Comment:
Good catch. Fixed.

Reply:
Btw in my solution I apply the same pattern as everywhere else however we can do better, but I don't want to improve the code in this PR, because it will probably be open for a while and will be rebased a lot.